### ISSUE TITLE
Fast typing with search auto-completion hangs Nyxt

### ISSUE THREAD

**Issue by aartaka** [atlas-engineer/nyxt#1929]
**Describe the bug**
I've just updated Nyxt to post-#1741 commits and noticed it hanging (becoming unresponsive, not rendering anything, I have to force-close the window). It seems to happen when typing into the `set-url` prompt.

**Precise recipe to reproduce the issue**
- Start Nyxt from any commit.
- Open `set-url` prompt.
- If your Nyxt is post-#1741 one, start typing random text really fast, erasing it and typing again.
- If your Nyxt is pre-#1741, input search engine prefix (default DDG should suffice) and do the same -- type random letters and erase them really fast. 
- Input some reasonable text, wait for one or two meaningful suggestions and repeat the same random typing.
- Try opening several links and repeat random typing.
- Nyxt should hang at some moment after 5-30 seconds of this random typing.

**Information**
- OS name+version: GuixSD, `guix describe`:
```
Generation 52	Dec 02 2021 21:57:28	(current)
  guix fab80c3
    repository URL: https://git.savannah.gnu.org/git/guix.git
    branch: master
    commit: fab80c36720bbdb85730708d0dc5fceabcc22a6e
  guix-gaming-games 43aa8af
    repository URL: https://gitlab.com/guix-gaming-channels/games
    branch: master
    commit: 43aa8af5f2bfbf4400fc46925909da56d4094cb7
  chur 8fcb5dd
    repository URL: https://github.com/aartaka/chur-guix
    branch: master
    commit: 8fcb5ddaa15fd10d71c1e50950405bb6d48e4393
  nonguix 971aa95
    repository URL: https://gitlab.com/nonguix/nonguix
    branch: master
    commit: 971aa95535df085050a8baa224f900c54ecd1e62
```
- Graphics card and driver: `lspci -v`:
```
...
00:02.0 VGA compatible controller: Intel Corporation 8th Gen Core Processor Gaussian Mixture Model (prog-if 00 [VGA controller])
	DeviceName: Onboard - Video
	Subsystem: Lenovo Device 312d
	Flags: bus master, fast devsel, latency 0, IRQ 131
	Memory at b0000000 (64-bit, non-prefetchable) [size=16M]
	Memory at a0000000 (64-bit, prefetchable) [size=256M]
	I/O ports at 3000 [size=64]
	Expansion ROM at 000c0000 [virtual] [disabled] [size=128K]
	Capabilities: <access denied>
	Kernel driver in use: i915
...
```
- Desktop environment / Window manager name+version: StumpWM 20.11.
- How you installed Nyxt (Guix pack, package manager, build from source): `guix package -f nyxt/build-scripts/nyxt.scm`
- Information from `show-system-information`:
```
Nyxt version: 2.2.2
Renderer version: GI-GTK
Operating system kernel: Linux 5.15.6
Lisp implementation: SBCL 2.1.9 (Dynamic space size: 3221225472)
Features: (:SLYNK :QUICKLISP :WEBKIT2 :WEBKIT2-2.34 :WEBKIT2-PASTE-PLAINTEXT
 :WEBKIT2-TRACKING :WEBKIT2-MUTE :WEBKIT2-EMOJI :WEBKIT2-MEDIA
 :WEBKIT2-SANDBOXING :GTK-3-22 :GTK-3-20 :GTK-3-18 :GTK-3-16 :GTK-3-14
 :GTK-3-12 :GTK-3-10 :GTK-3-8 :GTK-3-6 :GTK-3-4 :GTK :GDK-3-22 :GDK-3-20
 :GDK-3-18 :GDK-3-16 :GDK-3-14 :GDK-3-12 :GDK-3-10 :GDK-3-8 :GDK-3-6 :GDK-3-4
 :CAIRO-1-10 :CAIRO-1-12 :GDK-PIXBUF :GLIB-2-30 :GLIB-2-32 :GLIB-2-34
 :GLIB-2-36 :GLIB-2-38 :GLIB-2-40 :GLIB-2-42 :GLIB-2-44 :GLIB-2-46 :GLIB-2-48
 :GLIB-2-50 :GLIB-2-52 :GLIB-2-54 :GLIB-2-56 :GLIB-2-58 :GLIB :NYXT-2
 :FSET-EXT-STRINGS :CUSTOM-HASH-TABLE-NATIVE :SWANK :PLUMP-UTF-32 :GLOBAL-VARS
 :DECLARE-TYPES :PARENSCRIPT :SBCL+SAFE-STANDARD-READTABLE :NAMED-READTABLES
 :LPARALLEL :21BIT-CHARS :CHUNGA :FLEXI-STREAMS :CLOSER-MOP :CL-PPCRE-UNICODE
 :CL-UNICODE :CL-PPCRE :CL-JSON-DOUBLE-FLOAT-IS-SUBSUMED
 :CL-JSON-SINGLE-FLOAT-IS-SUBSUMED :BORDEAUX-THREADS :LPARALLEL.WITH-CLTL2
 :LPARALLEL.WITH-CAS :LPARALLEL.WITH-STEALING-SCHEDULER :SPLIT-SEQUENCE
 CHIPZ-SYSTEM:GRAY-STREAMS CFFI-FEATURES:FLAT-NAMESPACE CFFI-FEATURES:X86-64
 CFFI-FEATURES:UNIX :CFFI CFFI-SYS::FLAT-NAMESPACE ALEXANDRIA::SEQUENCE-EMPTYP
 :FAST-IO-SV :FAST-IO :SBCL-USES-SB-ROTATE-BYTE :CL-JSON-CLOS :CL-JSON
 :THREAD-SUPPORT :ASDF3.3 :ASDF3.2 :ASDF3.1 :ASDF3 :ASDF2 :ASDF :OS-UNIX
 :NON-BASE-CHARS-EXIST-P :ASDF-UNICODE :X86-64 :GENCGC :64-BIT :ANSI-CL
 :COMMON-LISP :ELF :IEEE-FLOATING-POINT :LINUX :LITTLE-ENDIAN
 :PACKAGE-LOCAL-NICKNAMES :SB-CORE-COMPRESSION :SB-LDB :SB-PACKAGE-LOCKS
 :SB-THREAD :SB-UNICODE :SBCL :UNIX)

ASDF version: 3.3.5
ASDF registries: (NYXT-SOURCE-REGISTRY ENVIRONMENT-SOURCE-REGISTRY USER-SOURCE-REGISTRY
                      USER-SOURCE-REGISTRY-DIRECTORY
                      DEFAULT-USER-SOURCE-REGISTRY SYSTEM-SOURCE-REGISTRY
                      SYSTEM-SOURCE-REGISTRY-DIRECTORY
                      DEFAULT-SYSTEM-SOURCE-REGISTRY)
Critical dependencies: (/gnu/store/z5dl8yb1slw3f34b3vnnv27m8digvfq6-cl-cffi-gtk-0.11.2-2.e9a46df/share/common-lisp/source/cl-cffi-gtk/gtk/cl-cffi-gtk.asd
 /gnu/store/wv106z80r2yxmh4irf877hnf4kxdpdnr-cl-gobject-introspection-0.3-1.d0136c8/share/common-lisp/source/cl-gobject-introspection/cl-gobject-introspection.asd
 /gnu/store/m586sbdszw4an40b8zbxdgck5pnmdh66-cl-webkit-3.4.0/share/common-lisp/source/cl-webkit/webkit2/cl-webkit2.asd)
```

**Comment by aartaka** [comment]
Again, this hang was there even before #1741, it's just that I only noticed it after using the current master for come time.

The reason is most probably some error in request/parsing completion functions that we don't muffle.

**Comment by jmercouris** [comment]
What is the error when ran from Slime?

**Comment by aartaka** [comment]
Ah, forget it. It was the typo in `renderer/gtk.lisp` that we left when transitioning from `load-status` to `status`. Should be fixed by 28c0803114e1203f5f8313a2c8620005c7eb5337.

**Comment by jmercouris** [comment]
I see, thanks :-)
